scripts/linters/custom_eslint_checks/rules/e2e-browser-nested-awaits.js

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Lint to ensure that e2e tests have nested awaits for
+   `await (await browser.switchTo().activeElement()).sendKeys(explanation)`
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: (
+        '`browser.switchTo().activeElement()` ' +
+        'should be wrapped in an `await` ' +
+        'expression before calling `sendKeys()`'),
+      category: 'Possible Errors',
+      recommended: true,
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      noNestedAwaits: (
+        'should have a nested `await` for ' +
+        '`browser.switchTo().activeElement()`')
+    },
+  },
+
+  create: function(context) {
+    /* Checks if this expression is syntactically the same as:
+       `await browser.switchTo().activeElement().sendKeys(<ARGS>)` */
+    var checkExprNode = function(node) {
+      node = node.expression;
+      if (node.type !== 'AwaitExpression' ||
+        node.argument.type !== 'CallExpression') {
+        return false;
+      }
+
+      node = node.argument.callee;
+      if (node.type !== 'MemberExpression' ||
+        node.property.type !== 'Identifier' ||
+        node.property.name !== 'sendKeys' ||
```

Comment:
This doesn't just apply to the `sendKeys` case (though that's the most common). It also applies to any other function call. You could check instead that we always `await` the result of `browser.switchTo().activeElement()`, regardless of whether there's an outer function call that we await

Sorry, the problem statement might not have been clear originally. The point here is that we always need to await the result of `browser.switchTo().activeElement()`. We often miss this when there's an additional function call like `.sendKeys()` and we only await that outer call.

Reply:
No problem, I misunderstood the original problem statement to refer to only the `sendKeys` case. This makes more sense to me though. I have changed the behavior of the lint check to match the description you've provided.